gitlabform/processors/group/group_settings_processor.py

```diff
@@ -1,12 +1,30 @@
+from logging import info, debug
+from typing import Dict
+
 from gitlabform.gitlab import GitLab
-from gitlabform.processors.single_entity_processor import SingleEntityProcessor
+from gitlab.v4.objects.groups import Group
+from gitlabform.processors.abstract_processor import AbstractProcessor
 
 
-class GroupSettingsProcessor(SingleEntityProcessor):
+class GroupSettingsProcessor(AbstractProcessor):
     def __init__(self, gitlab: GitLab):
-        super().__init__(
-            "group_settings",
-            gitlab,
-            get_method_name="get_group_settings",
-            edit_method_name="put_group_settings",
-        )
+        super().__init__("group_settings", gitlab)
+
+    def _process_configuration(self, group: str, configuration: Dict):
+        configured_group_settings = configuration.get("group_settings", {})
+
+        gitlab_group: Group = self.gl.get_group_by_path_cached(group)
+
+        if self._needs_update(gitlab_group.asdict(), configured_group_settings):
+            info("Updating group_settings  '%s'", gitlab_group.name)
+            self.update_group_settings(gitlab_group, configured_group_settings)
+        else:
+            debug("No update needed for Group Settings")
+
+    @staticmethod
+    def update_group_settings(gitlab_group: Group, group_settings_config: dict):
+        for key in group_settings_config:
+            value = group_settings_config[key]
+            debug(f"Updating setting {key} to value {value}")
+            gitlab_group.__setattr__(key, value)
+            gitlab_group.save()
```

Comment:
Instead of going through a loop, is it possible to use the `update()` method for a group and pass the entire config dictionary? I'm assuming this is available from python-gitlab library. Here's an example of how this is done for the "schedules processor".

https://github.com/gitlabform/gitlabform/blob/96a39eee118b6c7e00d6dd1d47d845213876ffb3/gitlabform/processors/project/schedules_processor.py#L86-L103

Would be good to follow a consistent pattern, if this makes sense. Not sure which way has more performance benefit 🤔 .

Reply:
I think calling `.update` would be best, _if_ it is available (sometimes py-gitlab docs are a bit confusing as to what is available where)

Reply:
I could not spot `update()` method on project or group. 
unless I see if thats available in api version 6 or 7 of gitlab.

Reply:
> sometimes py-gitlab docs are a bit confusing as to what is available where

Oh good. So I'm not the only one 😂 

> hi @amimas I could not spot update() method on project or group.

Thanks for checking @mahadevan-karthi-dwp . I have difficulty following python-gitlab docs. Not sure if I'm mis-reading this, but here it says the "GroupManager" is based on `CRUDMixin`, which  is based on `UpdateMixin` among others. Seems like `update()` method should be available.

https://python-gitlab.readthedocs.io/en/stable/api/gitlab.v4.html#gitlab.v4.objects.GroupManager

I'm still trying to get familiar with python-gitlab. So I could be wrong here. I think the important thing here is to make sure the entire setting/config gets passed to gitlab, which I think your loop does. I'm not sure if it will work if the config included nested dictionary, map, or array. Looking at GitLab's [update group setting api](https://docs.gitlab.com/ee/api/groups.html#update-group), there are couple of attributes that expects array type input (i.e. `unique_project_download_limit_allowlist`, `unique_project_download_limit_alertlist`). Maybe these don't work today and in that case your refactoring won't break anything that currently works. Would you be able to test it out or better yet... add a test case for it?

Anyways, I will leave it to @TimKnight-DWP as I know he's been working with python-gitlab lately. So he may have more insights. Thanks for helping with this refactoring.

Reply:
> > sometimes py-gitlab docs are a bit confusing as to what is available where
> 
> Oh good. So I'm not the only one 😂
> 
Omg, I’m

Reply:
@amimas @mahadevan-karthi-dwp  - I've previously tested this style loop and __set_attrs__ with an array input, which works perfectly, I've never tried it with a nested object/map etc because I've not seen one of those on GitLab